src/server.js

```diff
@@ -314,19 +314,26 @@ const PATH_NAME = process.env.NODE_ENV === "production" ?
   '~/gazelle-production/scripts/restartServers.sh' : process.env.NODE_ENV === "beta" ?
   '~/gazelle-beta/scripts/restartServers.sh' : __dirname + '/scripts/restartServers.sh';
 
+var isrestarted = false;
+
 editorTools.get('/restartserver', (req, res) => {
+
+  isrestarted = true;
+
   if (!process.env.NODE_ENV) {
-    // in dev mode
+    //in dev mode
     res.status(200).send("restarted");
     return;
   }
 
   const password = req.query.password;
+  
   if ((typeof password) !== "string" || password.length < 1) {
     res.status(401).send('invalid');
   }
   else if (hash(password) === 8692053) {
     exec(PATH_NAME, (err, stdout, stderr) => {
+      res.status(200).send('start');
```

Comment:
Hmmm, this is in the callback of the execution of the restartserver, the whole problem with the old approach is that this doesn't get called as the program gets terminated. You'll have to call this just before calling the restart server command. 

Also remember to set the `isrestarted` variable here.